pkg/controllers/job/helpers/helpers.go

```diff
@@ -45,6 +46,24 @@ func GetTaskIndex(pod *v1.Pod) string {
 	return ""
 }
 
+// ComparePodByIndex by pod index
+func ComparePodByIndex(lPod, rPod *v1.Pod) int {
+	if lPod == nil || rPod == nil {
+		return 0
```

Comment:
If any pod is `nil` or can not be conveted to integer, then returns `0`, is that reasonable?